### ISSUE TITLE
add option to automatically handle meta redirects [rt.cpan.org #95988]

### ISSUE THREAD

**Issue by oalders** [libwww-perl/libwww-perl#239]
Migrated from [rt.cpan.org#95988](https://rt.cpan.org/Ticket/Display.html?id=95988) (status was 'open')

Requestors:
* pagenyon@gmail.com

From pagenyon@gmail.com on 2014-05-26 22:21:39:
```
LWP already parses the head tag by default, and makes any contained meta http-equiv statements available as response headers.
I suggest adding an option (like meta_redirect => 1) that would tell LWP to automatically follow these redirects.

A web search shows this functionality has been asked for many times over the years, though it looks like no official feature request was made in RT or github. If there is a reason this will not be implemented, can you at least include an example in the docs on how the user can implement using request handlers.
```

From pagenyon@gmail.com on 2014-05-26 23:02:47:
```
On Mon May 26 18:21:39 2014, pagenyon wrote:
> LWP already parses the head tag by default, and makes any contained
> meta http-equiv statements available as response headers.
> I suggest adding an option (like meta_redirect => 1) that would tell
> LWP to automatically follow these redirects.
> 
> A web search shows this functionality has been asked for many times
> over the years, though it looks like no official feature request was
> made in RT or github. If there is a reason this will not be
> implemented, can you at least include an example in the docs on how
> the user can implement using request handlers.

figured out how to do with request handlers:

    $ua->set_my_handler(response_done => sub{
        my ($response, $ua, $handler) = @_;
        if (my ($refresh) = $res->remove_header('Refresh')) {
            $refresh =~ s/.*?\burl=//;
            my $uri = URI->new($refresh)->canonical;
            return if ref $uri eq 'URI::_generic';
            $response->code(301);
            $response->header(Location => $uri);
        }
    }, m_code => 200);
```

From ether@cpan.org on 2014-05-26 23:15:53:
```
On Mon, May 26, 2014 at 07:02:48PM -0400, pagenyon via RT wrote:
>             return if ref $uri eq 'URI::_generic';
>             $response->code(301);
>             $response->header(Location => $uri);

Checking for types like this is fragile -- the underscore is a hint that
you shouldn't be depending on this name.  Instead, check for the
capabilities that you need - e.g.  return unless $uri->can('scheme'),
or return unless $uri->some_property eq $value.
```

**Comment by grr** [comment]
is this something that would accepted if somebody implemented a better solution? i have something similar in my toolbox that could be adapted.

**Comment by oalders** [comment]
@grr yes, I think this would be helpful to have.

**Comment by oalders** [comment]
Closed via #409
